lib/index.js

```diff
@@ -808,6 +810,58 @@ GA.prototype.productClickedEnhanced = function(track) {
 };
 
 /**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListViewedEnhanced = function(track) {
+  var props = track.properties(); 
+  var products = track.products();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
+    window.ga('ec:addImpression', {
+      id: product.product_id,
+      category: track.category(),
+      list: props.list_id,
+      position: products.map(function(x) { return x.product_id; }).indexOf(product.product_id)
+    });
+  });
+  window.ga('send', 'pageview'); 
+  this.pushEnhancedEcommerce(track);
+};
+
+/**
+ * Add Impression - Enhanced Ecommerce
+ *
+ * https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#product-impression
+ *
+ * @api private
+ * @param {Track} track
+ */
+
+GA.prototype.productListFilteredEnhanced = function(track) {
+  var props = track.properties();
+  var products = track.products();
+  var filters = props.filters.map(function(obj) { return obj.type + ':' + obj.value;}).join();
+  var sorts = props.sorts.map(function(obj) { return obj.type + ':' + obj.value;}).join();
+  this.loadEnhancedEcommerce(track);
+  each(products, function(product) {
```

Comment:
Same as above, need to make sure to verify that either id (or sku fallback) or name are set - if neither, then skip, since GA will just reject the object.